js/widgets/temperament.js

```diff
@@ -2169,11 +2139,59 @@ class TemperamentWidget {
                 i -= 1;
             }
 
-            if (i <= pitchNumber && i >= 0 && this._playing && p < 2) {
+            if (i <= pitchNumber && i >= 0 && p < 2) {
                 setTimeout(() => {
-                    __playLoop(i);
+                    if (this.closed) {
+                        this.closed = !this.closed;
+                        return;
+                    } else if (!this._playing) {
+                        cell.innerHTML =
+                            '&nbsp;&nbsp;<img src="header-icons/' +
+                            "play-button.svg" +
+                            '" title="' +
+                            _("Play") +
+                            '" alt="' +
+                            _("Play") +
+                            '" height="' +
+                            TemperamentWidget.ICONSIZE +
+                            '" width="' +
+                            TemperamentWidget.ICONSIZE +
+                            '" vertical-align="middle" align-content="center">&nbsp;&nbsp;';
+                        if (i !== -1) {
+                            if (this.circleIsVisible == false && docById("wheelDiv4") == null) {
+                                updateNotesCircle(i - 1,TemperamentWidget.GREY);
+                                if (i == 11) {
+                                    //on completion of a full circle and on hitting '0' note in clockwise direction
+                                    updateNotesCircle(0,TemperamentWidget.GREY);
+                                } else if (i < 11) {
+                                    //in case of counter-clockwise direction, i.e., when this.playbackForward = false
+                                    updateNotesCircle(i + 1,TemperamentWidget.GREY);
+                                }
+                                this.notesCircle.refreshWheel();
+                            } else if (
+                                this.circleIsVisible == true &&
+                                docById("wheelDiv4") == null
+                            ) {
+                                docById("pitchNumber_" + (i - 1)).style.background =
+                                    platformColor.selectorBackground;
+                            } else if (docById("wheelDiv4") !== null) {
+                                this.wheel1.navItems[i - 1].fillAttr = TemperamentWidget.LIGHTGREY;
+                                this.wheel1.navItems[i - 1].sliceHoverAttr.fill =
+                                    TemperamentWidget.LIGHTGREY;
+                                this.wheel1.navItems[i - 1].slicePathAttr.fill =
+                                    TemperamentWidget.LIGHTGREY;
+                                this.wheel1.navItems[i - 1].sliceSelectedAttr.fill =
+                                    TemperamentWidget.LIGHTGREY;
```

Comment:
missed function call?

Reply:
No, actually the function `updateNotesCircle` works on `notesCircle`, whereas these lines use `wheel1` and this only happens once in the whole code. Thus, a different function would have to be made for `wheel1` which would not be that useful given that this function would have to be called just once.